File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/parking/GetParkingZoneUseCase.java
package com.lattis.ellipse.domain.interactor.parking;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.ParkingZone;
import com.lattis.ellipse.domain.repository.ParkingZoneRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by lattis on 24/05/17.
 */

public class GetParkingZoneUseCase extends UseCase {
    private int fleetId;
    private ParkingZoneRepository parkingZoneRepository;


    @Inject
    public GetParkingZoneUseCase(ThreadExecutor threadExecutor,
                                 PostExecutionThread postExecutionThread,
                                 ParkingZoneRepository parkingZoneRepository) {
        super(threadExecutor, postExecutionThread);
        this.parkingZoneRepository = parkingZoneRepository;
    }


    public GetParkingZoneUseCase withFleetID(int fleetId) {
        this.fleetId = fleetId;
        return this;
    }

    @Override
    protected Observable<List<ParkingZone>> buildUseCaseObservable() {
        return parkingZoneRepository.getParkingZone(fleetId);
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/parking/GetParkingZoneUseCase.kt
package com.lattis.domain.usecase.parking

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.ParkingRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.ParkingZone
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetParkingZoneUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val parkingRepository: ParkingRepository
) : UseCase<List<ParkingZone>>(threadExecutor, postExecutionThread) {
    private var fleetId = 0
    fun withFleetID(fleetId: Int): GetParkingZoneUseCase {
        this.fleetId = fleetId
        return this
    }

    override fun buildUseCaseObservable(): Observable<List<ParkingZone>> {
        return parkingRepository.getParkingZone(fleetId)
    }

}